### DISCUSSION TITLE
Easy way to change database on the fly

### DISCUSSION THREAD

**Discussion by MarijnRitzen** [laravel/framework#38664] [Ideas] [answered]
Hi, I am rebuilding an existent php website, and transferring it to laravel. The existing website used a database per tenant plus a main database. Right now I had to jump through a few loops to get it working, namely I had to add a 'connectToSecondDatabase' middleware that calls the config helper function to change the second database credentials. This kinda works, but I also had to add the middleware priority array to the Kernel.php file. It all feels a bit too hacky and forced for functionality that seems pretty  widely used (multi tenancy using a database per tenant). Couldn't laravel maybe implement something that would make this more easy?

**Comment by dennisprudlo** [comment, marked as answer]
[Tenancy for Laravel](https://tenancyforlaravel.com/) is a great package for implementing multi-tenancy with laravel.
